### PR TITLE
feat: Add multi-signature and multi-node support for signing and addi…

### PR DESCRIPTION
**Pull request by ivaylogarnev-limechain**
**Description**:
This PR adds support for multi-signature and multi-node transactions, enhancing the capabilities for signing and managing signatures.

**Related issue(s)**:
#2481 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by ivaylonikolov7** [comment]
You can add integration tests on the localnode network like we talked in DMs. Just sett the **first** node to something different than `AccountId(0,0,3)` and check if everything works as expected.

File: test/unit/EcdsaPrivateKey.js [hunk 1]
```diff
@@ -491,4 +493,120 @@ describe("EcdsaPrivateKey", function () {
...
+            sinon.assert.calledWith(
+                mockedTransaction.addSignature,
+                privateKey.publicKey,
+                mockedSignatures,
+            );
+
+            // Ensure that sign method of the privateKey._key was called the correct number of times
+            sinon.assert.callCount(privateKey._key.sign, 3);
+        });
+    });
```
Comment (ivaylonikolov7) [inline test/unit/EcdsaPrivateKey.js:611]: I wonder if we should create a new `PrivateKey` file with these unit tests in it. It just feels weird that we have this `PrivateKey` tests in the `EcdsaPrivateKey`.

**Comment by ivaylonikolov7** [comment]
Please check in integration tests if this works for the `FileAppend` transaction as it has chunked transactions.

**Review by agadzhalov — APPROVED** [review]
LGTM

**Review by ivaylonikolov7 — APPROVED** [review]

**Pull request merged**